python_tests/test_python_frontend.py

```diff
@@ -1387,25 +1387,61 @@ def test_compute_contiguity(self):
         self.assertEqual(compute_contiguity(sizes, strides), contiguity)
 
     def test_compute_tensor_descriptor(self):
-        sizes = [2, 1, 3, 1, 4, 3]
-        strides = [12, 4, 4, 4, 1, 0]
-        contiguity = [True, None, True, None, True, None]
-        stride_order = [5, 4, 3, 2, 1, 0]
-        computed_contiguity, computed_stride_order = compute_tensor_descriptor(
-            sizes, strides
-        )
-        self.assertEqual(computed_contiguity, contiguity)
-        self.assertEqual(computed_stride_order, stride_order)
-
-        sizes = [2, 3, 1, 5, 4]
-        strides = [28, 4, 14, 0, 1]
-        contiguity = [False, None, True, True, None]
-        stride_order = [4, 2, 3, 0, 1]
-        computed_contiguity, computed_stride_order = compute_tensor_descriptor(
-            sizes, strides
+        configs = (
+            (
+                # size
+                [2, 1, 3, 1, 4, 3],
+                # stride
+                [12, 4, 4, 4, 1, 0],
+                # expected contiguity
+                [True, None, True, None, True, None],
+                # expected stride_order
+                [5, 4, 3, 2, 1, 0],
+            ),
+            (
+                [2, 3, 1, 5, 4],
+                [28, 4, 14, 0, 1],
```

Comment:
Is the third dim counted as contiguous even though its stride is 14 instead of 12 because the size is one?

Reply:
I think there's a confusion here.

The `expected contiguity` flag is marked with the sorted stride. That's also how we set contiguity flag in c++ API, regarding alloc_domain.

So our alloc_domain is permuted regarding the stride_order. i.e.
sorted size:    [2, 1, 3, 5, 4]
sorted stride:  [28, 14, 4, 0, 1]
expected contiguity flag: [ False, None, True, None, True]

broadcast dimension with size 1 and stride 14 is marked as `None`, since it's a broadcast here.

Reply:
Oh of course. My mistake. Thanks for the explanation.